### ISSUE TITLE
[MEDIUM] Update authentication APIs to enable HybridApp auth refactor

### ISSUE THREAD

**Issue by Julesssss** [Expensify/App#53556]
## Problem

I am working with @war-in and @staszekscp on a [HybridApp authentication refactor](https://github.com/Expensify/App/issues/49844). 

They are making good progress but require some backend changes in order to complete the work. 

## Solution

- [x] Update backend APIs to return tryNewDot NVP data in authentication APIs (`BeginSignIn`, `SignUpUser`, `SignInUser`, plus redirection and SSO APIs)
- [x] Rename the api response key (`tryNewDot`) which is confusingly named to `tryNewDotRedirectDismissed` in Web and Auth 

> we localized three more commands which need to return tryNewDot value
> - SignInWithGoogle
> - SignInWithApple
> - SignInWithShortLivedAuthToken (for SAML)
> We can unify SignInWithShortLivedAuthToken and SinginUser by moving value returned by SignInUser to BeginSignIn.
We can also simplify the frontend code and return nvp_tryNewDot in the onyxData (like we do for e.g. credentials in BeginSignIn command) so those onyx updates would be handled automatically.

**Onyx Data**
```
{
  "onyxData": [
    {
      "onyxMethod": "merge",
      "key": "nvp_tryNewDot",
      "value": {
        "classicRedirect": {
          "dismissed": true/false
        }
      }
    }
  ],
  "jsonCode": 200,
  "requestID": "8ecc27b03b9abbcf-WAW"
}
```

**Comment by Julesssss** [comment]
Finishing up other HybridApp tasks, I should have time for this soon after the new year

**Comment by Julesssss** [comment]
Still waiting for the ability to prioritise

**Comment by Julesssss** [comment]
The front-end work is in progress.

**Comment by Julesssss** [comment]
@war-in asked me to provide these changes. I'll be picking up this issue soon.

**Comment by Julesssss** [comment]
THis is on my todo list after a fire, and one other HybridApp backend issue.

**Comment by Julesssss** [comment]
Aiming to work on this this week!

**Comment by Julesssss** [comment]
I have been held up by change in priorities, this is still on my roadmap after we ship HybridApp NewDot signup and redirect to classic tasks.

**Comment by Julesssss** [comment]
OKAY. I am not going to jynx myself this time...

@war-in, is the desired format under `Onyx Data` still correct? If so I'll start by refactoring the hardcoded `tryNewDot` response data into a valid onyx payload.

**Comment by war-in** [comment]
@Julesssss I think it's fine to return it in regular onyx update data but maybe let's keep it in both places for now? 🤔 We can remove it then once we confirm its fine

**Comment by Julesssss** [comment]
Sounds good to me 👍

**Comment by Julesssss** [comment]
Okay, we're returning the full NVP data in `SigninUser`: 

<img width="322" alt="Image" src="https://github.com/user-attachments/assets/243abab7-0390-4fea-91b0-76b1f26cc467" />

Same for `SignUpUser` user: 

<img width="406" alt="Image" src="https://github.com/user-attachments/assets/9eee29b9-4a8e-4a6c-a0fd-6c251b94bc5c" />

**Comment by Julesssss** [comment]
Next week I need to figure out why `tryNewDot` is now `true`

**Comment by Julesssss** [comment]
> Next week I need to figure out why tryNewDot is now always returning true.

Bug fixed. PR is ready for review, I just need to fix a remaining test case failure.

**Comment by Julesssss** [comment]
> So now those commands should return NVP, right?

@war-in yeah, the full tryNewDot NVP data will be returned within onyxData for all of these cases. In addition we'll return `tryNewDotDismissed`. 

I'll deprecate `tryNewDot` as it is confusingly named, but for now I am returning that too:

<img width="673" alt="Image" src="https://github.com/user-attachments/assets/cf3eb993-08bb-4479-9310-d17fb49f7afb" />

**Comment by Julesssss** [comment]
To avoid duplicated onyx data I had to make a large refactor and update automated tests. That's all done now and the final step is to see if I can make the NVP retrieval more efficient.

**Comment by Julesssss** [comment]
@war-in could you made do without the returned `tryNewDotRedirecDismissed` param from `SignInWithShortLivedAuthToken`? I have the onyxData being returned but I would need to take extra time to make that available. I can do that later if it's necessary.

**Comment by Julesssss** [comment]
The changes are working well, but add some extra Auth calls for retrieving the NVP data. As this isn't ideal for performance I will raise an Auth PR to improve this. It'll delay this PR by a few days but will be worth the wait.

**Comment by Julesssss** [comment]
Finished up some improvements and test fixes. After dealing with a flakey test I have put the PR in initial review.

**Comment by Julesssss** [comment]
Hey @war-in, the following commands are now returning onyxData on staging and some will give you `tryNewDotDismissed` in the response: 

- BeginSignIn
- SignInUser
- SignUpUser
- SignInWithGoogle
- SignInWithApple
- SignInWithShortLivedAuthToken (SAML / redirect from OldDot)

Let me know if you need anything else at this point. `SignInWithShortLivedAuthToken`, `SignInWithGoogle`, and `SignInWithApple` won't give you `tryNewDotDismissed` but I should be able to figure this out if you need it.

**Comment by Julesssss** [comment]
I'm going to close this issue. We can use the [parent issue](https://github.com/Expensify/App/issues/49844) for the rest of the changes.

**Comment by Julesssss** [comment]
Reopening while the [remaining](https://github.com/Expensify/Auth/pull/14762) Auth keys are refactored.

**Comment by Julesssss** [comment]
Auth [PR](https://github.com/Expensify/Auth/pull/14762) updated, will ship for review if tests pass

**Comment by Julesssss** [comment]
Auth PR deployed, closing
